src/modules/webRTC/components/Offerrings.tsx

```diff
@@ -0,0 +1,103 @@
+import { Box } from "../../components/Box";
+import { Button } from "../../components/Button";
+import { useCallback, useRef, useState } from "react";
+import { ShouldRender } from "../../components/ShouldRender";
+import { TextArea } from "../../components/TextArea";
+import { Loader } from "../../components/Loader";
+import { webRTC_connection } from "../baseWebRTC";
+import { handleCopy } from "../utils/handleCopy";
+import { buttonStyle, textAreaStyle } from "./styles";
+
+type Props = { webRTCConnection: webRTC_connection | undefined }
+
+export const Offerings = ({ webRTCConnection }: Props) => {
+
+    const textAreaOffer = useRef<HTMLTextAreaElement>(null);
+    const textAreaAnswer = useRef<HTMLTextAreaElement>(null);
+ 
+    const [showOffer, setShowOffer] = useState<boolean>(true);
+    const [answerProcessed, setAnswerProcessed] = useState<boolean>(false);
+    const [offerIsLoading, setOfferIsLoading] = useState<boolean>(false);
+    const [offer, setOffer] = useState<string>("")
+    
+    const createOffer = useCallback(async() => {
+        console.info('create Offer...');   
```

Comment:
with logging we either just move all of it to utils, or better yet get rid of most of it and use class decorators for debug mode. We've discussed this before so I won't go into details